### PR TITLE
Marker not updated properly fixed

### PR DESCRIPTION
**Pull request by Santhosh-Sellavel**
@timszot 
### Details
The issue is specifically occurs in Mobile devices & mWeb

### Fixed Issues

$ #4357 

### Tests & QA Steps
1. Open new Expensify app
2. Wait for any new message on the chat list page
3. When a new message is received open the chat
4. Should show `NEW` indicator above the new messages
5. Go back to the chat list
6. Open the same chat again, `NEW` indicator should not be shown.
7.  Now go back to the chat list, & wait until a new message from the same user
8. Open the same chat now
9. Should show `NEW` indicator above the new messages
10. Go back to chat list
11. Open the same chat again, `NEW` indicator should not be shown.

### Tested On
- [x] Mobile Web
- [x] iOS
- [x] Android

#### Mobile Web
https://user-images.githubusercontent.com/85645967/132128854-5560109e-b8f8-4bff-a151-426969fc1163.mp4

#### iOS
https://user-images.githubusercontent.com/85645967/132128985-71615ca1-0ebf-46a0-806f-35048d3c53e3.mov

#### Android
https://user-images.githubusercontent.com/85645967/132129234-ddf16cb1-58d1-4f06-b343-8b4acce4fd5f.mov

### CODE REVIEW

**Comment by Santhosh-Sellavel** [comment]
@sketchydroide can review my changes if possible. I see @timszot is OOO until sep 7.

I'll update screenshot/videos after an initial review thanks!

**Comment by sketchydroide** [comment]
Is this still a WIP?

**Comment by Santhosh-Sellavel** [comment]
> Is this still a WIP?
No added WIP
 - For screenshots/videos only 
 - @timszot first review

**Comment by sketchydroide** [comment]
It is still a draft, that's what I meant

**Comment by Santhosh-Sellavel** [comment]
> It is still a draft, that's what I meant
Yes mate

**Comment by deetergp** [comment]
@Santhosh-Sellavel WIP = Work in Progress, which was how we designated draft PRs before Github has the ability to mark a PR as a draft.

**Review by deetergp — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ReportActionsView.js [hunk 1]
```diff
@@ -149,10 +141,8 @@ class ReportActionsView extends React.Component {
             return true;
         }
 
-        // If the new marker has changed places (because the user manually marked a comment as Unread), we have to
-        // update the component.
-        if (nextProps.report.newMarkerSequenceNumber > 0
-            && nextProps.report.newMarkerSequenceNumber !== this.props.report.newMarkerSequenceNumber) {
+        // If the new marker has changed places so update the component.
```
[the remark this answers is not in the document]
Comment (Santhosh-Sellavel) [reply src/pages/home/report/ReportActionsView.js:144]: Sure

File: src/pages/home/report/ReportActionsView.js [hunk 2]
```diff
@@ -373,6 +360,19 @@ class ReportActionsView extends React.Component {
         updateLastReadActionID(this.props.reportID);
     }
 
+    /**
+     * Updates NEW marker position
+     */
+
```
[the remark this answers is not in the document]
Comment (Santhosh-Sellavel) [reply src/pages/home/report/ReportActionsView.js:366]: Will do.

**Comment by Santhosh-Sellavel** [comment]
cc: @sketchydroide @deetergp Ready for review

File: src/pages/home/report/ReportActionsView.js [hunk 3]
```diff
@@ -216,15 +206,12 @@ class ReportActionsView extends React.Component {
...
-
-        // We want to mark the unread comments when user resize the screen to desktop
-        // Or user move back to report from LHN
-        if (shouldRecordMaxAction && (
+        } else if (shouldRecordMaxAction && (
             prevProps.isDrawerOpen !== this.props.isDrawerOpen
             || prevProps.isSmallScreenWidth !== this.props.isSmallScreenWidth
         )) {
             updateLastReadActionID(this.props.reportID);
+            this.updateUnreadIndicatorPosition(this.props.report.unreadActionCount);
```
Comment (sketchydroide) [inline src/pages/home/report/ReportActionsView.js:214]: weird spacing here

**Review by sketchydroide — APPROVED** [review]

**Review by deetergp — APPROVED** [review]
Code looks good and tests pass 👍

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-09-16.  :confetti_ball:

**Pull request merged**